commit bbe25c462fc7abf41746e5d91ec0e6ecdca082e5
Author: Gawain Lynch <gawain.lynch@gmail.com>
Date:   Sun Apr 5 15:53:36 2015 +0100

    [OAuth2] Custom function to try a second approach for GitHub email
    addresses when required

diff --git a/lib/League/OAuth2/Provider/Github.php b/lib/League/OAuth2/Provider/Github.php
index 8182a4e..3fe2e06 100644
--- a/lib/League/OAuth2/Provider/Github.php
+++ b/lib/League/OAuth2/Provider/Github.php
@@ -41,12 +41,32 @@ class Github extends AbstractProvider
         return $this->domain.'/api/v3/user/emails';
     }
 
+    private function reQueryEmail(AccessToken $token)
+    {
+        $emails = $this->getUserEmails($token);
+
+        if (empty($emails)) {
+            return null;
+        }
+
+        foreach ($emails as $email) {
+            if ($email['primary']) {
+                return $email;
+            }
+        }
+    }
+
     public function userDetails($response, AccessToken $token)
     {
         $user = new User();
 
         $name = (isset($response->name)) ? $response->name : null;
-        $email = (isset($response->email)) ? $response->email : null;
+
+        if (isset($response->email) && isset($response->email)) {
+            $this->reQueryEmail($token);
+        } else {
+            $email = (isset($response->email)) ? $response->email : null;
+        }
 
         $user->exchangeArray([
             'uid' => $response->id,
